ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -224,6 +224,13 @@ public class RNTrackPlayer: RCTEventEmitter, AudioPlayerDelegate {
     
     @objc(add:before:resolver:rejecter:)
     public func add(trackDicts: [[String: Any]], before trackId: String?, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
+        
+        //tell app to receive remote control events on add
+        //so we can remove it on remove
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
What would be the downside of doing this on `play`?

Reply:
This is just giving the app the ability to send events to MPNowPlayingInfoCenter. When you add tracks, you are basically _intending_ on playing, imo. I didn't have this on play,  because this functionality isn't necessarily doing anything except for giving the app the ability to update the `MPNowPlayingInfoCenter`. Play will handle the updates to the info center. On `reset`, it'll clear everything, and take away access so the InfoCenter clears. Then, you'll need to add more tracks anyway, so it'll re-initialize and prepare for utilizing it again

Reply:
It also requires the main thread, so I'm trying to limit the amount of calls as little as necessary